File: LeetCode/python/maximumOddBinaryNumber.py
def maximumOddBinaryNumber(s):
    listStr = list(s)
    left = 0
    for i in range(len(listStr)):
        if listStr[i] == '1':
            listStr[i], listStr[left] = listStr[left], listStr[i]
            left+=1
    print(''.join(listStr))
     
    listStr[left-1], listStr[len(listStr)-1] = listStr[len(listStr) -1], listStr[left-1]
    print(''.join(listStr))
    return ''.join(listStr)
    
    
maximumOddBinaryNumber("010")

File: LeetCode/java/MaximumOddBinaryNumber.java
public class MaximumOddBinaryNumber {
    public static void main(String[] args) {
        String s = new String("010");
        var l = s.toCharArray();

        int left = 0;
        int length = s.length();
        for (int i = 0; i < length; i++) {
            if (l[i] == '1') {
                swap(l, i, left);
                left++;
            }
        }
        swap(l, left-1, length-1);

    }

    static public char[] swap(char[] s, int i, int j) {
        var temp = s[j];
        s[j] = s[i];
        s[i] = temp;
        return s;
    }
}
